ui/src/app/layout.tsx

```diff
@@ -20,13 +21,15 @@ const RootLayout = ({
 }: Readonly<{
     children?: React.ReactNode;
 }>) => (
-    <html lang="en">
-        <body className={sourceSans3.variable}>
-            <Navbar />
-            <main className="h-full 2xl:min-h-[72.3vh]">{children}</main>
-            <Footer />
-        </body>
-    </html>
+        <html lang="en">
+            <body className={sourceSans3.variable}>
+                <Navbar />
+                <ErrorBoundary>
+                <main className="h-full 2xl:min-h-[72.3vh]">{children}</main>
+                </ErrorBoundary>
```

Comment:
I found it nicer to just have <ErrorBoundary /> so that the error modal displays without removing the main content.